flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStreamSink.java

```diff
@@ -35,9 +35,9 @@
 
 	private final SinkTransformation<T> transformation;
 
-	@SuppressWarnings("unchecked")
 	protected DataStreamSink(DataStream<T> inputStream, StreamSink<T> operator) {
-		this.transformation = new SinkTransformation<T>(inputStream.getTransformation(), "Unnamed", operator, inputStream.getExecutionEnvironment().getParallelism());
```

Comment:
Unrelated changes, please revert or pull into a separate hotfix if you deem it necessary (I don't).

Reply:
I will revert it.